test/src/test/java/org/corfudb/runtime/concurrent/CorfuQueueTxTest.java

```diff
@@ -24,7 +25,22 @@
 @Slf4j
 public class CorfuQueueTxTest extends AbstractTransactionsTest {
     @Override
-    public void TXBegin() { OptimisticTXBegin(); }
+    public void TXBegin() {
+        OptimisticTXBegin();
+    }
+
+    public void TXBegin(TransactionType type) {
+        switch (type){
+            case WRITE_AFTER_WRITE:
+                WWTXBegin();
+                return;
+            case OPTIMISTIC:
+                OptimisticTXBegin();
+                return;
+            default:
+                throw new IllegalArgumentException("Unsupported TXN type");
```

Comment:
Do we want to include input "type" as parameter in exception for debugging?

Reply:
I was thinking of this, but this is really used for this test case in this file only, so skipped it for simplicity.

Reply:
but let me make the change as you suggest. thanks